.buildkite/pipeline.yml

```diff
@@ -0,0 +1,17 @@
+steps:
+  - label: "Swift Build and Tests"
+    command:
+      - "swift test --parallel"
+      - "swift build --configuration release"
```

Comment:
Does the `swift build` depend on or benefit from the `swift test` running before it?

I'm wonder if this should be two separate pipeline steps, with the `swift build` depending on `swift test` (example below to show what I mean).

That said, maybe the provisioning time / dependency downloads / build caches etc from the `swift test` benefit the `swift build` step enough that it's worth keeping them as a single step. I'm happy either way!

I don't really see much _benefit_ in what I'm suggesting, and it has more YAML and more machine setup/teardown so feel free to ignore it :) Just thinking out loud.

```yaml
steps:

  - label: "Swift Tests"
    key: swift-test
    command: "swift test --parallel"
    branches:
      - "main"
      - "buildkite-runner-migration"
    agents:
      os: "macos"
    env:
      DEVELOPER_DIR: "/Applications/Xcode_{{matrix.xcode_version}}.app/Contents/Developer"
    matrix:
      setup: 
        xcode_version:
          - "15.2"
          - "14.3.1"

  - label: "Swift Build"
    key: swift-build
    depends_on: swift-test
    command: "swift build --configuration release"
    branches:
      - "main"
      - "buildkite-runner-migration"
    agents:
      os: "macos"
    env:
      DEVELOPER_DIR: "/Applications/Xcode_{{matrix.xcode_version}}.app/Contents/Developer"
    matrix:
      setup: 
        xcode_version:
          - "15.2"
          - "14.3.1"
```

Reply:
Yeah I have thought this too, it provides great separation, the only thing would be this might kick off 4 tasks in parallel which might be too much? and you are right we'd want to the test pass before build, but anyway great one 👍

Reply:
The `depends_on: swift-test` would ensure that neither of the Build jobs would begin before the Test jobs complete. (But is that a real dependency? I'm not sure).

Reply:
It should work, let me have a try. you guys have way more expertise than I do in setting up pipeline 😄 so TIL

Reply:
![image](https://github.com/buildkite/test-collector-swift/assets/30752103/b5dae2b0-05e6-45fa-95ad-317bb1721ee1)

Reply:
Looks good, and the Waterfall view shows that the two parts ran sequentially 👍🏼

<img width="1212" alt="image" src="https://github.com/buildkite/test-collector-swift/assets/15759/584e2766-8652-4254-83bf-64c848e2f22c">